sycl/test-e2e/NewOffloadDriver/diamond_shape.cpp

```diff
@@ -0,0 +1,107 @@
+// RUN: %{build} %{embed-ir} -O2 --offload-new-driver -o %t.out
+// RUN: %{run} %t.out
+
+// Test complete fusion with private internalization specified on the
+// accessors for a combination of four kernels, forming a diamond-like shape and
+// repeating one of the kernels. This test uses the new offloading model for
+// linking device objects.
+
+#include <sycl/detail/core.hpp>
+#include <sycl/ext/codeplay/experimental/fusion_wrapper.hpp>
+#include <sycl/properties/all_properties.hpp>
+
+using namespace sycl;
+
+struct AddKernel {
+  accessor<int, 1> accIn1;
+  accessor<int, 1> accIn2;
+  accessor<int, 1> accOut;
+
+  void operator()(id<1> i) const { accOut[i] = accIn1[i] + accIn2[i]; }
+};
+
+int main() {
+  constexpr size_t dataSize = 512;
+  int in1[dataSize], in2[dataSize], in3[dataSize], tmp1[dataSize],
+      tmp2[dataSize], tmp3[dataSize], out[dataSize];
+
+  for (size_t i = 0; i < dataSize; ++i) {
+    in1[i] = i * 2;
+    in2[i] = i * 3;
+    in3[i] = i * 4;
+    tmp1[i] = -1;
+    tmp2[i] = -1;
+    tmp3[i] = -1;
+    out[i] = -1;
+  }
+
+  queue q{ext::codeplay::experimental::property::queue::enable_fusion{}};
```

Comment:
is it ok to not check that fusion is supported by device? do we expect it to always be available?

queue impl has code:
if (MPropList.has_property<
            ext::codeplay::experimental::property::queue::enable_fusion>() &&
        !MDevice->get_info<
            ext::codeplay::experimental::info::device::supports_fusion>()) {
      throw sycl::exception(
          make_error_code(errc::invalid),
          "Cannot enable fusion if device does not support fusion");

Reply:
That's a good point, the tests in the `KernelFusion` subdirectory are implicitly guarded by the local lit configuration: 
https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/KernelFusion/lit.local.cfg

So a `REQUIRES: fusion` would be needed at the minimum.